### ISSUE TITLE
The GRU/LSTM function bugs when using TF2.7

### ISSUE THREAD

**Issue by hayatejp** [tensorflow/tensorflow#53701]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):binary
- TensorFlow version (use command below):2.7
- Python version:3.8
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:11.5
- GPU model and memory:NVIDIA Geforce 1070, 8GB

**Describe the current behavior**

Hello，When I used the GRU/LSTM model to accomplish the text classification with IMDB dataset，there were some problems  in model  fitting process.This is the dataset shape information.
![datasets](https://user-images.githubusercontent.com/4904955/148642698-a1fe2d86-ed76-404e-a89b-34a122a21618.png)
When I set the GRU layer with  return_sequences=True:

![GRU](https://user-images.githubusercontent.com/4904955/148642812-1fd832e2-7d8f-4bc4-94bc-443f0ed4677a.png)

![1-True](https://user-images.githubusercontent.com/4904955/148642730-781a87a5-67f1-426f-a3a3-30e70698967c.png)

Then I removed the return_sequences=False,the code was seemed to be normal,but the gradient was't decreased:

![1-false](https://user-images.githubusercontent.com/4904955/148642765-bffba379-bd36-4d8d-bcc0-1667afff676c.png)

The LSTM performance was also similar with the GRU. If I used the two layers GRU with the last GRU's return_sequences=False，the performance of model was same as the single GRU/LSTM layer.
![M-RNN](https://user-images.githubusercontent.com/4904955/148642917-8adb0da9-b3d3-4a96-8000-ac2764170bdf.png)

![M-RNN2](https://user-images.githubusercontent.com/4904955/148642905-e30f9f6b-4483-4bc5-b4a9-5b9dc396da92.png)

**Describe the expected behavior**
I asked my teammates who used **tf2.6.2**,they said all the problem I mentioned above didn't occur when they ran these codes. But some people who used  **tf2.7** found the same problem. I also checked the different systems like linux in colab or docker,it seems that the bugs could happen when using tf2.7.I has updated my code with zip attachment,

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

The Code is here:
[imdb.zip](https://github.com/tensorflow/tensorflow/files/7833147/imdb.zip)

**Comment by mohantym** [comment]
Hi @hayatejp !
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by hayatejp** [comment]
Hi mohantym ,I will post the issue on Keras,Thank you for your support

**Comment by hayatejp** [comment]
> Hi @hayatejp ! Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) To know more see; https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999 . Thanks!

Hi mohantym ,I will post the issue on Keras,Thank you for your support

**Comment by mohantym** [comment]
Ok @hayatejp ! Can you please close this issue here as it will be tracked in keras repo. Thanks!

**Comment by hayatejp** [comment]
> Ok @hayatejp ! Can you please close this issue here as it will be tracked in keras repo. Thanks!

OK~
